### PR TITLE
change zeroclipboard to clipboardjs

### PR DESCRIPTION
**Pull request by Preovaleo**
solve #2047

### CODE REVIEW

**Comment by Preovaleo** [comment]
i make a mistake i fix it

File: js/share.js [hunk 1]
```diff
@@ -36,14 +36,14 @@ function share(dest) {
...
     var clipboard = new Clipboard('#copylink', {
         text: function (trigger) {
-            return $(trigger).select();
+            return $(trigger).val();
         }
     });
-
+    
     clipboard.on('success', function (e) {
+        $('#copylink').select();
```
Comment (Preovaleo) [inline js/share.js]: I add this to keep the input content selected when we release the left button

File: .gitignore [hunk 2]
```diff
@@ -50,3 +50,4 @@ Desktop.ini
 
 # Mac crap
 .DS_Store
+/nbproject/
```
Comment (LeoColomb) [inline .gitignore]: Could you explain this addition?

File: js/share.js [hunk 3]
```diff
@@ -36,16 +36,14 @@ function share(dest) {
...
-	});
-};                     
-
+    var clipboard = new Clipboard('#copylink', {
+        text: function (trigger) {
+            return $(trigger).val();
+        }
+    });
+    
+    clipboard.on('success', function (e) {
```
Comment (LeoColomb) [inline js/share.js]: `e` isn't used.

File: .gitignore (hunk 2 above)
Comment (Preovaleo) [inline .gitignore]: sure this is some of my IDE config files (NetBeans)

File: js/share.js (hunk 3 above)
Comment (Preovaleo) [inline js/share.js]: from https://clipboardjs.com/

```
var clipboard = new Clipboard('.btn');

clipboard.on('success', function(e) {
    console.info('Action:', e.action);
    console.info('Text:', e.text);
    console.info('Trigger:', e.trigger);

    e.clearSelection();
});

clipboard.on('error', function(e) {
    console.error('Action:', e.action);
    console.error('Trigger:', e.trigger);
});
```

(e) could contain some info but do you want to log the success?

Comment (LeoColomb) [inline js/share.js]: As you want. That said, console logging is not recommended.
If not, you should not add the `e` parameter.

File: .gitignore [hunk 4]
```diff
@@ -50,3 +50,6 @@ Desktop.ini
 
 # Mac crap
 .DS_Store
+
+# NetBeans files
```
Comment (LeoColomb) [inline .gitignore]: :+1: :ok_hand:

File: js/share.js [hunk 5]
```diff
@@ -42,7 +42,7 @@ function init_clipboard() {
         }
     });
     
-    clipboard.on('success', function (e) {
+    clipboard.on('success', function () {
```
Comment (LeoColomb) [inline js/share.js]: :ok_hand:

**Comment by ozh** [comment]
(for the record, still havent got time to test this, but this is not forgotten ;)

**Comment by Preovaleo** [comment]
up ?

**Comment by ozh** [comment]
Sorry about the lack of reply, still havent got time to properly test and merge this...

**Comment by ozh** [comment]
Actually took time to give this PR the attention it deserves, I thought there would be more edge cases to test but the changes are pretty straightforward and everything seems to work flawlessly (FF, Chrome, IE). Thanks a bunch !! :bow:
